hw/xbox/nv2a/nv2a_pgraph.c

```diff
@@ -299,26 +299,81 @@ static uint64_t fnv_hash(const uint8_t *data, size_t len);
 static uint64_t fast_hash(const uint8_t *data, size_t len, unsigned int samples);
 
 /* PGRAPH - accelerated 2d/3d drawing engine */
+
+static uint32_t pgraph_rdi_read(PGRAPHState *pg,
+                                unsigned int select, unsigned int address)
+{
+    uint32_t r = 0;
+    switch(select) {
+    case RDI_INDEX_VTX_CONSTANTS0:
+    case RDI_INDEX_VTX_CONSTANTS1:
+        assert((address / 4) < NV2A_VERTEXSHADER_CONSTANTS);
+        r = pg->vsh_constants[address / 4][3 - address % 4];
```

Comment:
If RDI always sends 4 words in reverse order, maybe do: `address ^= 0x3` in the beginning, or even `unsigned int part = address % 4; address /= 4`?

I might run a unit test for this in the next days.

Reply:
Got no time for this at the moment. Code can remain as-is and improved later.